network-mux/network-mux.cabal

```diff
@@ -33,7 +33,7 @@ library
                        binary          >=0.8 && <0.11,
                        bytestring      >=0.10 && <0.11,
                        containers      >=0.5 && <0.7,
-                       network         >=3.1 && <3.2,
+                       network         >=3.1 && <3.1.2,
```

Comment:
@newhoggy What's the reason for this change? Does the new 3.1.2 not work somehow? It seems to compile just fine for me.

Reply:
It doesn't build on Windows.

See https://github.com/haskell/network/issues/484

Reply:
I see. We have many dependencies on the `network` package. I'm wondering whether it would make more sense to add a constraint to `cabal.project` instead of changing the constraint in *one* `.cabal` file. @coot Your opinion?

Reply:
Actually, you're right, the upper bound should be here, because it is indeed only `network-mux` that relies on the function that went missing. And if we put the constraint in `cabal.project`, then all downstream repos should add that same constraint to their `cabal.project`, instead of automatically inheriting it through the `network-mux.cabal`.